src/Database/LSMTree/Normal.hs

```diff
@@ -167,8 +167,9 @@ import           Database.LSMTree.Internal.Normal
 -- such instances in use at once.
 type TableHandle :: (Type -> Type) -> Type -> Type -> Type -> Type
 data TableHandle m k v blob = forall h. Typeable h =>
-    TableHandle (Internal.TableHandle m h)
+    TableHandle {-# UNPACK #-} !(Internal.TableHandle m h)
```

Comment:
The `UNPACK` pragma goes through newtypes here, so you are unpacking the `RWVar` fields into here?

I'm always a bit cautious about `UNPACK`. Without looking into the Core I wouldn't be sure it's an improvement, i.e. there is no function being called down the line that takes an internal handle or `RWVar` for which it needs to be reboxed.

Reply:
I've removed the `UNPACK` for now. We can add them back later if we think it's worth it